File: Linguagem C/1021.c
#include <stdio.h>
#include <locale.h>
float n, notas[7], moedas[7];


int main(){
	scanf("%f",&n);
		while(n < 0 || n > 1000000.00){
			scanf("%f",&n);
		}
	
	
	
		while( n >= 100){
			n-=100;
			notas[0]++; 
		}
		while( n >= 50){
			n-=50;
			notas[1]++; 
		}
		while( n >= 20){
			n-=20;
			notas[2]++; 
		}
		while( n >= 10){
			n-=10;
			notas[3]++; 
		}
		while( n >= 5){
			n-=5;
			notas[4]++; 
		}
		while( n >= 2){
			n-=2;
			notas[5]++; 
		}
		while( n >= 1){
			n-=1;
			moedas[0]++; 
		}
		while( n >= 0.50){
			n -=0.50;
			moedas[1]++;
		}
		while( n >= 0.25){
			n -=0.25;
			moedas[2]++;
		}
		while( n >= 0.10){
			n-=0.10;
			moedas[3]++;
		}
		while( n >= 0.05){
			n-=0.05;
			moedas[4]++;
		}
		moedas[5] = n;
		moedas[5] *=100;
		
		
		printf("NOTAS:\n");
		printf("%.0f nota(s) de R$ 100.00\n",notas[0]);
		printf("%.0f nota(s) de R$ 50.00\n",notas[1]);
		printf("%.0f nota(s) de R$ 20.00\n",notas[2]);
		printf("%.0f nota(s) de R$ 10.00\n",notas[3]);
		printf("%.0f nota(s) de R$ 5.00\n",notas[4]);
		printf("%.0f nota(s) de R$ 2.00\n",notas[5]);
		
		printf("MOEDAS:\n");
		printf("%.0f moeda(s) de R$ 1.00\n",moedas[0]);
		printf("%.0f moeda(s) de R$ 0.50\n",moedas[1]);
		printf("%.0f moeda(s) de R$ 0.25\n",moedas[2]);
		printf("%.0f moeda(s) de R$ 0.10\n",moedas[3]);
		printf("%.0f moeda(s) de R$ 0.05\n",moedas[4]);
		printf("%.0f moeda(s) de R$ 0.01\n",moedas[5]);
		
	
	return 0;
}

File: Linguagem Java/1021.java
package pkg1021;

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner input = new Scanner (System.in);
        
        int notas[] = new int[12];
        float n = input.nextFloat();
            while(n <= 0 || n >= 1000000.00){
                n = input.nextFloat();
            }
            
        while(n >= 100){
            notas[0]++;
            n -= 100.0f;
        }while(n >= 50){
            notas[1]++;
            n -= 50.0f;
        }while(n >= 20){
            notas[2]++;
            n -= 20.0f;
        }while(n >= 10){
            notas[3]++;
            n -= 10.0f;
        }while(n >= 5){
            notas[4]++;
            n -= 5.0f;
        }while(n >= 2){
            notas[5]++;
            n -= 2.0f;
        }
        
        
        System.out.println("NOTAS:");
        System.out.println(notas[0]+" nota(s) de R$ 100.00");
        System.out.println(notas[1]+" nota(s) de R$ 50.00");
        System.out.println(notas[2]+" nota(s) de R$ 20.00");
        System.out.println(notas[3]+" nota(s) de R$ 10.00");
        System.out.println(notas[4]+" nota(s) de R$ 5.00");
        System.out.println(notas[5]+" nota(s) de R$ 2.00");
        
        while(n >= 1){
            notas[6]++;
            n--;
        }while(n >= 0.5f){
            notas[7]++;
            n -= 0.5f;
        }while(n >= 0.25f){
            notas[8]++;
            n -= 0.25f;
        }while(n >= 0.10f){
            notas[9]++;
            n -= 0.10f;
        }while(n >= 0.05f){
            notas[10]++;
            n -= 0.05f;
        }
        n *= 100;
            while(n > 0.1f){
                notas[11]++;
                n -= 1;
            }
        
        System.out.println("MOEDAS:");
        System.out.println(notas[6]+" moeda(s) de R$ 1.00");
        System.out.println(notas[7]+" moeda(s) de R$ 0.50"); 
        System.out.println(notas[8]+" moeda(s) de R$ 0.25");
        System.out.println(notas[9]+" moeda(s) de R$ 0.10");
        System.out.println(notas[10]+" moeda(s) de R$ 0.05");
        System.out.println(notas[11]+" moeda(s) de R$ 0.01");
    }
    
}
